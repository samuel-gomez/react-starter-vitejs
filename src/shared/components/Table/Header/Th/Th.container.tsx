import { ReactNode } from 'react';
import ThSortableContainer, { TsetSort } from './ThSortable';
import Th from './Th';

export type TThContainer = Omit<TsetSort, 'onSort' | 'field'> & {
  ThSortableCmpt?: typeof ThSortableContainer;
  ThCmpt?: typeof Th;
  children?: ReactNode;
  classModifier?: string;
  onSort?: TsetSort['onSort'];
  field?: TsetSort['field'];
};

const ThContainer = ({ field, sorting, onSort, ThSortableCmpt = ThSortableContainer, ThCmpt = Th, ...rest }: TThContainer) =>
  field && onSort ? <ThSortableCmpt {...rest} sorting={sorting} onSort={onSort} field={field} /> : <ThCmpt {...rest} />;

export default ThContainer;
