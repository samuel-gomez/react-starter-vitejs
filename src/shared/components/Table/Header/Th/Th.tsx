import { ReactNode } from 'react';
import withClassNameModifier from 'shared/hoc/WithClassNameModifier';

export type TTh = {
  className?: string;
  children?: ReactNode;
  ariaLabel?: string;
  role?: string;
};

const DEFAULT_CLASSNAME = 'af-table__th';

const Th = withClassNameModifier(
  ({ className, children, role = 'columnheader', ...rest }: TTh) => (
    <th className={className} role={role} {...rest}>
      {children}
    </th>
  ),
  DEFAULT_CLASSNAME,
);

export default Th;
