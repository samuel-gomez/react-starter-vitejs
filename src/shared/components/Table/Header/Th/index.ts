export { default } from './Th.container';
export type { TThContainer } from './Th.container';
export type { TsetSort, Torder } from './ThSortable';
