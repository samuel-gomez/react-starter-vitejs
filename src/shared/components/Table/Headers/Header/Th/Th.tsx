import { Table as TableTk } from '@axa-fr/react-toolkit-all';
import type { ReactNode } from 'react';
import ThSortableContainer, { type TsetSort } from './ThSortable';

export type TTh = Omit<TsetSort, 'onSort' | 'field'> & {
  ThSortableCmpt?: typeof ThSortableContainer;
  ThCmpt?: typeof TableTk.Th;
  children?: ReactNode;
  classModifier?: string;
  onSort?: TsetSort['onSort'];
  field?: TsetSort['field'];
  scope?: string;
};

const Th = ({ field, sorting, onSort, ThSortableCmpt = ThSortableContainer, ThCmpt = TableTk.Th, scope, ...rest }: TTh) =>
  field && onSort ? <ThSortableCmpt scope={scope} sorting={sorting} onSort={onSort} field={field} {...rest} /> : <ThCmpt scope={scope} {...rest} />;

export default Th;
