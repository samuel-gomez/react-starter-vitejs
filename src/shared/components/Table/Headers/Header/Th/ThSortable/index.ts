export { default } from './ThSortable.container';
export type { TThSortableContainer, Torder, TsetSort } from './ThSortable.container';
export { default as ThSortable } from './ThSortable';
export type { TThSortable } from './ThSortable';
