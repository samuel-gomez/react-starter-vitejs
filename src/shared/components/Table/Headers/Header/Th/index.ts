export { default } from './Th';
export type { TTh } from './Th';
export type { TsetSort, Torder } from './ThSortable';
