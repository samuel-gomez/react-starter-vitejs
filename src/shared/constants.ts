import packageJson from '../../package.json';

/** ***********************************************************************
 * FORM CONSTANTS
 ************************************************************************ */
export const MSG_REQUIRED = 'Le champ est obligatoire';

export const MSG_LENGTH_3 = 'Le champ doit faire au moins 3 caractères';
export const MSG_FORMAT = 'Format invalide';
export const MSG_ERROR_FORMAT = 'Format de fichier incorrect';

export const DEFAULT_OPTION_LABEL = '- Sélectionner -';
export const DEFAULT_OPTION = {
  value: '',
  label: DEFAULT_OPTION_LABEL,
};

// patterns
export const patternEmail = /^[\w-+]+@([\w-+]+\.)+[\w-]{2,}$/;
export const patternPhoneNumber = /^[+0][\d ]*$/;
export const patternZipCode = /^\d{5}$/;
export const patternSiret = /^\d{14}$/;
export const patternSiren = /^\d{9}$/;
export const patternSiretSiren = /^\d{9,14}$/;

export const phoneNumberRule = {
  required: true,
  minLength: 4,
  maxLength: 20,
  pattern: patternPhoneNumber,
};

export const emailRule = {
  required: true,
  minLength: 6,
  maxLength: 60,
  pattern: patternEmail,
};

/** ***********************************************************************
 * API STATUS CONSTANTS
 ************************************************************************ */
export const STATUS_HTTP = {
  SUCCESS: 200,
  CREATED: 201,
  BAD_REQUEST: 400,
  UNAUTHORIZE: 401,
  FORBIDDEN: 403,
  NOTFOUND: 404,
  SERVER_ERROR: 500,
};

export const STATUS_API = {
  SUCCESS: STATUS_HTTP.SUCCESS,
  CREATED: STATUS_HTTP.CREATED,
  WARNING: 40,
  ERROR: 50,
};

/** ***********************************************************************
 * API URL
 ************************************************************************ */
export const API_URL = {
  BASE: 'base',
  GITHUB: 'github',
};

export const STATUS_HTTP_MESSAGES = {
  [STATUS_HTTP.SUCCESS]: 'Succès: La requête a réussi',
  [STATUS_HTTP.CREATED]: 'Succès: Création réussie',
  [STATUS_HTTP.BAD_REQUEST]: 'Erreur: La syntaxe de la requête est erronée',
  [STATUS_HTTP.UNAUTHORIZE]: 'Erreur: Une authentification est nécessaire pour accéder à la ressource',
  [STATUS_HTTP.FORBIDDEN]: 'Erreur: Accès non autorisé',
  [STATUS_HTTP.NOTFOUND]: 'Erreur: Elément non trouvé',
  [STATUS_HTTP.SERVER_ERROR]: 'Erreur: Problème technique ! Contacter votre support',
};

/** ***********************************************************************
 * USER CONSTANTS
 ************************************************************************ */

export const ADMIN = 'Admin';
export const READER = 'Reader';
export const WRITER = 'Writer';
export const USER = 'User';
export const ALL = '';

export const PROFILS = [ADMIN, READER, WRITER, USER, ALL];

/** ***********************************************************************
 * COMMONS CONSTANTS
 ************************************************************************ */

export const MODIFIER_CLASS = {
  success: 'success',
  disabled: 'disabled',
};

/** ***********************************************************************
 * VERSION CONSTANT
 ************************************************************************ */

const VERSION_TOOLKIT = 'master';

/** ***********************************************************************
 * DEMO CONSTANTS
 ************************************************************************ */

export const STORYBOOK = 'https://axafrance.github.io/react-toolkit/latest/storybook/?path=/story/';
export const DESIGN_SYSTEM = 'https://axafrance.github.io/design-system/';
export const GITHUB = `https://github.com/AxaGuilDEv/react-toolkit/tree/v${packageJson.dependencies['@axa-fr/react-toolkit-all']}/packages/`;
export const GITHUB_API = `AxaGuilDEv/react-toolkit/${VERSION_TOOLKIT}/packages/`;
