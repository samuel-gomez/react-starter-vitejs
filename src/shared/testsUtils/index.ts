export * from './customRender';
export { default as emptyFunction } from '../helpers/emptyFunction';
export { default as MOCK_API_URL } from './constants';
export { default as renderWithContainer } from './renderWithContainer';
