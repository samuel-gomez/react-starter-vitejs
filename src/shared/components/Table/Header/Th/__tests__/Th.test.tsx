import { describe, it, expect } from 'vitest';
import { renderWithContainer } from 'shared/testsUtils';
import Th from '../Th';

const container = document.createElement('tr');

describe('Th', () => {
  it('Render <Th/>', () => {
    const { baseElement } = renderWithContainer(<Th>child th</Th>, container);
    expect(baseElement).toMatchSnapshot();
  });

  it('Render <Th/> with modifier', () => {
    const { baseElement } = renderWithContainer(<Th classModifier="custom">child th</Th>, container);
    expect(baseElement).toMatchSnapshot();
  });
});
