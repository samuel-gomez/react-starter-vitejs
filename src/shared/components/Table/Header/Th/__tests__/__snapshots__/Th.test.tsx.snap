// Vitest Snapshot v1

exports[`Th > Render <Th/> 1`] = `
<tr>
  <th
    class="af-table__th"
    role="columnheader"
  >
    child th
  </th>
</tr>
`;

exports[`Th > Render <Th/> with modifier 1`] = `
<tr>
  <th
    class="af-table__th af-table__th--custom"
    role="columnheader"
  >
    child th
  </th>
</tr>
`;
